internal/graph/cached_resolver.go

```diff
@@ -139,13 +139,8 @@ func (c *CachedCheckResolver) ResolveCheck(
 	ctx context.Context,
 	req *ResolveCheckRequest,
 ) (*ResolveCheckResponse, error) {
-	ctx, span := tracer.Start(ctx, "ResolveCheck", trace.WithAttributes(
-		attribute.String("store_id", req.GetStoreID()),
-		attribute.String("resolver_type", "CachedCheckResolver"),
-		attribute.String("tuple_key", req.GetTupleKey().String()),
-		attribute.Bool("is_cached", false),
-	))
-	defer span.End()
+	span := trace.SpanFromContext(ctx)
+	span.SetAttributes(attribute.Bool("is_cached", false))
```

Comment:
Do we only want to set this when we know is_cached != true (i.e. line 163)?  Otherwise, would it be possible that line 157 fails because the attribute is already set (but to false)

Reply:
`SetAttributes` overrides values:

https://github.com/open-telemetry/opentelemetry-go/blob/9656d0afa72646101e859bf8a1c6d05b73ee094d/trace/trace.go#L382-L385

Reply:
> Do we only want to set this when we know is_cached != true 

I have no strong feelings about this, we could do either!